rsocket/RSocketResponder.cpp

```diff
@@ -70,35 +70,51 @@ RSocketResponder::handleRequestChannelCore(
     }
 
     void onError(folly::exception_wrapper ex) noexcept override {
-      DCHECK(inner_);
-      inner_->onError(std::move(ex));
+      VLOG(3) << "handleRequestChannelCore::onError: " << ex.what();
+      std::lock_guard<std::mutex> guard(errorMutex_);
+      if (inner_) {
+        inner_->onError(std::move(ex));
 
-      inner_.reset();
-      subscription_.reset();
+        inner_.reset();
+        subscription_.reset();
+        error_.reset();
```

Comment:
This `error_.reset()` shouldn't be necessary if we do `onError(std::move(error_))` in `subscribe()`.